dtfs-central-api/src/v1/validation/utilisation-report-service/selected-fee-record-validator.test.ts

```diff
@@ -37,11 +51,114 @@ describe('selected fee record validator', () => {
       // Act + Assert
       expect(() => validateSelectedFeeRecordsAllHaveSamePaymentCurrency(selectedFeeRecords)).toThrow(InvalidPayloadError);
     });
+
+    function aFeeRecordWithPaymentCurrency(paymentCurrency: Currency) {
+      return FeeRecordEntityMockBuilder.forReport(aReconciliationInProgressReport()).withPaymentCurrency(paymentCurrency).build();
+    }
+  });
+
+  describe('validateThatAllSelectedFeeRecordsWithPaymentsFormACompletePaymentGroup', () => {
+    const feeRecordFindBySpy = jest.spyOn(FeeRecordRepo, 'findByIdWithPaymentsAndFeeRecords');
+
+    describe(`when the first fee record has status ${FEE_RECORD_STATUS.TO_DO}`, () => {
+      it('should not throw', async () => {
+        // Arrange
+        const selectedFeeRecords = [
+          FeeRecordEntityMockBuilder.forReport(aReconciliationInProgressReport()).withId(1).withStatus(FEE_RECORD_STATUS.TO_DO).build(),
+        ];
+        feeRecordFindBySpy.mockResolvedValue(selectedFeeRecords);
```

Comment:
I'm a little surprised why we don't reset spy implementation after/before each on this file -- could leak into other test cases in this file?

Reply:
Missed this, my bad, added an `afterEach` to reset after